components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -332,92 +227,8 @@ protected boolean validateAudience(String currentAudience, List<String> audience
                 return true;
             }
         }
-        return logAndReturnFalse("None of the audience values matched the tokenEndpoint Alias: " + currentAudience);
-    }
-
-
-    /**
-     * @deprecated use @{@link RequestObjectValidatorImpl#getX509CertOfOAuthApp(String, String)}} instead
-     * to retrieve the public certificate of the Service Provider in X509 format.
-     */
-    @Deprecated
-    protected Certificate getCertificateForAlias(String tenantDomain, String alias) throws RequestObjectException {
-        return getX509CertOfOAuthApp(alias, tenantDomain);
-    }
-
-    /**
-     * Get the X509Certificate object containing the public key of the OAuth client.
-     *
-     * @param clientId clientID of the OAuth client (Service Provider).
-     * @param tenantDomain tenant domain of Service Provider.
-     * @return X509Certificate object containing the public certificate of the Service Provider.
-     */
-    protected Certificate getX509CertOfOAuthApp(String clientId, String tenantDomain) throws RequestObjectException {
-
-
-        try {
-            return OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
-        } catch (IdentityOAuth2Exception e) {
-            String errorMsg = "Error retrieving application certificate of OAuth app with client_id: " + clientId +
-                    " , tenantDomain: " + tenantDomain;
-            if (StringUtils.isNotBlank(e.getMessage())) {
-                // We expect OAuth2Util.getX509CertOfOAuthApp() to throw an exception with a more specific reason for
-                // not being able to retrieve the X509 Cert of the service provider.
-                errorMsg = e.getMessage();
-            }
-            throw new RequestObjectException(errorMsg, e);
-        }
-    }
-
-    /**
-     * Validate the signedJWT signature with given certificate
-     *
-     * @param signedJWT
-     * @param x509Certificate
-     * @return
-     */
-    protected boolean isSignatureVerified(SignedJWT signedJWT, Certificate x509Certificate) {
```

Comment:
can't remove protected methods, you can only deprecate

Reply:
These are moved to `org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil` as discussed keep the common functions of `RequestObjectValidator` and `CIBARequestObjectValidator` in the same place

Reply:
still you can't change any public/protected APIs from last released version of the product. You can only deprecate them and use the newer method underlying

Reply:
Readded and deprecated with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1743/commits/01427042989ea11ebcfc9313d4ebf007aac7a19b